uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/apis/leave-request-api.js

```diff
@@ -157,7 +157,11 @@ define([
         this.sendPOST('LeaveRequest', 'isValid', params)
           .then(function (data) {
             if (data.count > 0) {
-              deferred.reject(data.values);
+              var errorMessage = _.map(data.values, function (value) {
+                return _.isArray(value) ? value.join('</br>') : JSON.stringify(value);
```

Comment:
you are doing something related to the presentation layer (putting every message in a new line), which something that *should not* happen on the api service, as it shouldn't care how the messages will be displayed on the page. Leave the job to the controller, or even better to the template